pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook_test.go

```diff
@@ -655,95 +496,12 @@ func TestHandleMemberCluster(t *testing.T) {
-					Name: "test-mc",
-					Object: runtime.RawExtension{
-						Raw: fleetMCObjectBytes,
-					},
-					UserInfo: authenticationv1.UserInfo{
-						Username: "test-user",
-						Groups:   []string{"system:masters"},
-					},
-					RequestKind: &utils.MCMetaGVK,
-					Operation:   admissionv1.Create,
-				},
-			},
-			resourceValidator: fleetResourceValidator{
-				decoder: decoder,
-			},
-			wantResponse: admission.Allowed(fmt.Sprintf(validation.ResourceAllowedFormat, "test-user", utils.GenerateGroupString([]string{"system:masters"}), admissionv1.Create, &utils.MCMetaGVK, "", types.NamespacedName{Name: "test-mc"})),
-		},
-		"deny create fleet MC for non system user": {
-			req: admission.Request{
-				AdmissionRequest: admissionv1.AdmissionRequest{
-					Name: "test-mc",
-					Object: runtime.RawExtension{
-						Raw: fleetMCObjectBytes,
-					},
-					UserInfo: authenticationv1.UserInfo{
-						Username: "test-user",
-						Groups:   []string{"test-group"},
-					},
-					RequestKind: &utils.MCMetaGVK,
-					Operation:   admissionv1.Create,
-				},
-			},
-			resourceValidator: fleetResourceValidator{
-				decoder: decoder,
-			},
-			wantResponse: admission.Denied(fmt.Sprintf(validation.ResourceDeniedFormat, "test-user", utils.GenerateGroupString([]string{"test-group"}), admissionv1.Create, &utils.MCMetaGVK, "", types.NamespacedName{Name: "test-mc"})),
-		},
-		"allow create upstream MC for non system user": {
-			req: admission.Request{
-				AdmissionRequest: admissionv1.AdmissionRequest{
-					Name: "test-mc",
-					Object: runtime.RawExtension{
-						Raw: mcObjectBytes,
-					},
-					UserInfo: authenticationv1.UserInfo{
-						Username: "test-user",
-						Groups:   []string{"test-group"},
-					},
-					RequestKind: &utils.MCMetaGVK,
-					Operation:   admissionv1.Create,
-				},
-			},
-			resourceValidator: fleetResourceValidator{
-				decoder: decoder,
-			},
-			wantResponse: admission.Allowed("upstream member cluster resource is allowed to be created/deleted by any user"),
-		},
-		"allow delete fleet MC for user in system:masters group": {
-			req: admission.Request{
-				AdmissionRequest: admissionv1.AdmissionRequest{
-					Name: "test-mc",
-					Object: runtime.RawExtension{
-						Raw: fleetMCObjectBytes,
-					},
-					OldObject: runtime.RawExtension{
-						Raw: fleetMCObjectBytes,
-					},
-					UserInfo: authenticationv1.UserInfo{
-						Username: "test-user",
-						Groups:   []string{"system:masters"},
-					},
-					RequestKind: &utils.MCMetaGVK,
-					Operation:   admissionv1.Delete,
-				},
-			},
-			resourceValidator: fleetResourceValidator{
-				decoder: decoder,
-			},
-			wantResponse: admission.Allowed(fmt.Sprintf(validation.ResourceAllowedFormat, "test-user", utils.GenerateGroupString([]string{"system:masters"}), admissionv1.Delete, &utils.MCMetaGVK, "", types.NamespacedName{Name: "test-mc"})),
-		},
-		"deny delete fleet MC for non system user": {
+		"deny update of fleet MC spec by non whitelisted user": {
```

Comment:
I wonder why this case is kept while others are pruned?

Reply:
There cases are back-doors we added for white-listed users which are not required to be tested by E2Es